code/game/objects/items/plushes.dm

```diff
@@ -35,6 +35,68 @@
 	name = "ratvar plushie"
 	desc = "An adorable plushie of the clockwork justiciar himself with new and improved spring arm action."
 	icon_state = "plushvar"
+	var/obj/item/toy/plush/narplush/clash_target
+
+/obj/item/toy/plush/plushvar/Initialize()
+	. = ..()
+	START_PROCESSING(SSobj, src)
+
+/obj/item/toy/plush/plushvar/Destroy()
+	STOP_PROCESSING(SSobj, src)
+	. = ..()
+
+/obj/item/toy/plush/plushvar/process()
+	if(clash_target)
+		return
+	var/obj/item/toy/plush/narplush/P = locate() in range(1, src)
+	if(P && istype(P.loc, /turf/open))
+		clash_of_the_plushies(P)
+
+/obj/item/toy/plush/plushvar/proc/clash_of_the_plushies(obj/item/toy/plush/narplush/P)
+	clash_target = P
+	say("YOU.")
+	P.say("Ratvar?!")
+	var/obj/item/toy/plush/a_winnar_is
+	var/victory_chance = 10
+	for(var/i in 1 to 10) //We only fight ten times max
+		if(!src || !P)
+			return
+		if(!Adjacent(P))
+			visible_message("<span class='warning'>The two plushies angrily flail at each other before giving up.</span>")
+			clash_target = null
+			return
+		playsound(src, 'sound/magic/clockwork/ratvar_attack.ogg', 50, TRUE, frequency = 2)
+		sleep(2.4)
```

Comment:
timers instead of sleeps

Reply:
That's a great deal of timers, and there'd need to be a string of different procs to do it. Would `set waitfor = FALSE` not be a cleaner solution to do the same thing?

Reply:
BTW the reason he wants timers instead of set waitfor = FALSE is because of an unrelated project